src/Builtins.jl

```diff
@@ -309,3 +309,105 @@ function show(io::IO, ::MIME"text/html", radio::Radio)
 end
 
 get(radio::Radio) = radio.default
+
+
+"""Generate Table of Contents using Markdown cells. Headers h1-h6 are used. 
+
+`title` is the custom Title header to this element. 
+`indent` aligns header elements in a heirarchy.
+
+# Examples
+`@bind TableOfContents()`
+
+`@bind TableOfContents(indent=True)`
+
+`@bind TableOfContents(title="Table of Contents")`
+
+"""
+struct TableOfContents
+    title::AbstractString
+    indent::Bool
+end
+TableOfContents(;title::AbstractString="", indent::Bool=false) = TableOfContents(title, indent)
+
+function show(io::IO, ::MIME"text/html", toc::TableOfContents)
+
+    if length(toc.title) > 0
+        print(io, """<div class="title">$(toc.title)</div>""")
+    end
+
+    withtag(io, :script) do
+        print(io, """
+            const elementsOfType = (type) => Array.from(
+                document.querySelectorAll(
+                    "pluto-notebook pluto-output " + type
+                )
+            ).map(el => {
+                const parentCellId = function(el) {
+                    while (el.nodeName != 'PLUTO-CELL') {
+                        el = el.parentNode;
+                        if (!el) return null;
+                    }
+                    return el.id;
+                }            
+                return {
+                    "el": el,
+                    "parentCellId": parentCellId(el)
+                }
+            })
+            
+            const plutoCellIds = Array.from(
+                document.querySelectorAll(
+                    "pluto-notebook pluto-cell"
+                )
+            ).map(el => el.id)
+            
+            const headers = [...elementsOfType("h1"), ...elementsOfType("h2"),...elementsOfType("h3"),...elementsOfType("h4"), ...elementsOfType("h5"),...elementsOfType("h6")] 
```

Comment:
Would it make sense to have the allowed headers as input argument? E.g. the user would provide the level depth they require (1 to 6) and only headers up to this depth are used in the ToC?

If LaTeX has any value here, I would argue that the default depth should be 3 (because in latex by default you get section numbering up to e.g. `2.2.3`)

Reply:
(but this is a minor comment - please don't let it hold this PR back)

Reply:
Added the depth limit. ✅
Tempted to do auto numbering, but wouldn't it be jarring? a section name would have a number in the TOC, but not in the actual location. I'd rather let user for numbering the section.

Reply:
I don't know how the plugin of Jupyter does it, but the autonumbering there is reflected in _both_ the TOC and the headers. So the headers will literally display 1.1 Header. But maybe this is too invasive or just too hard to do from the design of Pluto?

(to be honest I actually liked the auto numbering)

Reply:
Yes, updating the user created content itself would be invasive & fraught with assumptions.
Lets see the feedback on auto-numbering.

Reply:
I've checked Jupyter again and they only change the display. So if I have markdown sections:
```
# Title
## Subtitle

# Next title
```
they will be _displayed_ as "1. Title, 1.1 Subtitle, 2. Next title", but when I double click the markdown cells to edit them, the numbers are not there. So it is only what displayed that is altered and thus I would call it non-invasive.

Reply:
Header cells would need to store the numbering in their state making them *special markdown cells*. Adding this _hidden state_ from an external PlutoUI addon seems invasive 🤔